scalyr_agent/util.py

```diff
@@ -89,6 +91,8 @@
 Original error: %s
 """.strip()
 
+RFC3339_STR_REGEX = re.compile(r"(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})")
```

Comment:
Can you add a comment above with an example string this matches.  I find that better grounds long regex

Reply:
Also interestingly, for python3,  `\d` will match all Unicode digits (python2 `\d` only matches ascii by default) and I wonder if that will affect performance at all.

Reply:
@czerwingithub I've added a comment in fae292682a03b96d5d7884590c0341c983898934.

Based on the benchmark results, it doesn't affect it. In fact, both versions (with and without strptime) are a bit faster under Python 3.

I also just tested it by adding ``re.ASCII`` flag and it appears to be tiny bit faster (~0.1 us, on the side of the statistical error) than the version without it.

Reply:
I tried yet another implementation which utilizes ``string.split`` which appears to be another couple of percents faster than regex one.

As per the code comment - I intentionally didn't assign split values to temporary variables (aka unpack it into tuple) which would make it more readable because it's slightly slower (and means we utilize a bunch of temporary very short lived variables).

Python 3 - regex approach:

![Screenshot from 2020-04-08 23-35-37](https://user-images.githubusercontent.com/125088/78836294-1df71500-79f2-11ea-865b-e7aa13fab80e.png)

Python 2 - regex approach:

![Screenshot from 2020-04-08 23-36-16](https://user-images.githubusercontent.com/125088/78836286-19caf780-79f2-11ea-9a7f-4d36ec21cb75.png)

Python 3 - string split approach

![Screenshot from 2020-04-09 12-16-43](https://user-images.githubusercontent.com/125088/78885048-326ff780-7a5c-11ea-9841-332c48ae93cd.png)


Python 2 - string split approach

![Screenshot from 2020-04-09 12-17-18](https://user-images.githubusercontent.com/125088/78885056-34d25180-7a5c-11ea-9581-e7de56da80f0.png)